packages/migrate/src/__tests__/DbPull.test.ts

```diff
@@ -427,19 +429,51 @@ describe('common/sqlite', () => {
     expect(ctx.mocked['process.stdout.write'].mock.calls.join('\n')).toMatchInlineSnapshot(`
 
 
-                                                            - Introspecting based on datasource defined in prisma/invalid.prisma
+                                                                        - Introspecting based on datasource defined in prisma/invalid.prisma
 
-                                                            ✔ Introspected 3 models and wrote them into prisma/invalid.prisma in XXXms
-                                                                  
-                                                            Run prisma generate to generate Prisma Client.
+                                                                        ✔ Introspected 3 models and wrote them into prisma/invalid.prisma in XXXms
+                                                                              
+                                                                        Run prisma generate to generate Prisma Client.
 
-                                        `)
+                                                `)
     expect(ctx.mocked['process.stderr.write'].mock.calls.join('\n')).toMatchInlineSnapshot(``)
 
     expect(ctx.fs.read('prisma/invalid.prisma')).toMatchSnapshot()
   })
 })
 
+describe('postgresql - missing database', () => {
+  const defaultConnectionString =
+    process.env.TEST_POSTGRES_URI_MIGRATE || 'postgres://prisma:prisma@localhost:5432/tests-migrate'
+
+  // replace database name, e.g., 'tests-migrate', with 'unknown-database'
+  const connectionString = defaultConnectionString.split('/').slice(0, -1).join('/') + '/unknown-database'
+
+  test('basic introspection --url', async () => {
+    const introspect = new DbPull()
+    const result = introspect.parse(['--print', '--url', connectionString])
+    await expect(result).rejects.toThrowErrorMatchingInlineSnapshot(`
+
+      P1003 The introspected database does not exist: postgres://prisma:prisma@localhost:5432/unknown-database
```

Comment:
Unrelated to this PR I guess: I'm not a fan of printing the user and password to the logs 😨 
I'll create an issue for this

Reply:
It's only for --url, still undocumented so it's not a blocker

Reply:
I think we already handle secretation elsewhere. Perhaps this should be a function defined just once in the Rust engines and exported via Wasm. Can you please open an issue for this?

Reply:
https://github.com/prisma/prisma/issues/17286